[maven] using an url as file does not work on windows.

- using an url as file does not work on windows.
- [ maven 3.2.4 ] gradle or maven version
- [1.0.0.BETA4  ] spotless version
- [ windows 10 - 1709] operating system and version
```
				<configuration>
					<java>
						<eclipse>
							<file>https://git.mycompany.de/projects/INFRA/repos/pm-codestyle/raw/eclipse-java-pm-style.xml</file>
							<version>4.7.1</version>
						</eclipse>
					</java>
</configuration>
```
```
[ERROR] Failed to execute goal com.diffplug.spotless:spotless-maven-plugin:1.0.0.BETA4:apply (code-formatting) on project project-api: Execution code-formatting of goal com.diffplug.spotless:spotless-maven-plugin:1.0.0.BETA4:apply failed: Unable to create tem
poraty file 'https://git.mycompany.de/projects/INFRA/repos/pm-codestyle/raw/eclipse-java-pm-style-1527576190601.xml' in the output directory: Failed to create directory D:\git\project\api\target\https:\git.mycompany.de\projects\INFRA\repos\pm-codestyle\raw -> [
Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.diffplug.spotless:spotless-maven-plugin:1.0.0.BETA4:apply (code-formatting) on project project-api: Execution code-formatting of goal com.diffplug.spotless:spotless-maven-plugi
n:1.0.0.BETA4:apply failed: Unable to create temporaty file 'https://git.mycompany.de/projects/INFRA/repos/pm-codestyle/raw/eclipse-java-pm-style-1527576190601.xml' in the output directory
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:224)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:120)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:347)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:154)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:582)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:214)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:158)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution code-formatting of goal com.diffplug.spotless:spotless-maven-plugin:1.0.0.BETA4:apply failed: Unable to create temporaty file 'https://git.mycompany.de/projects/INFRA/repos/pm-codestyle/raw/ec
lipse-java-pm-style-1527576190601.xml' in the output directory
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:143)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
        ... 19 more
Caused by: java.lang.RuntimeException: Unable to create temporaty file 'https://git.mycompany.de/projects/INFRA/repos/pm-codestyle/raw/eclipse-java-pm-style-1527576190601.xml' in the output directory
        at com.diffplug.spotless.maven.FileLocator.locateFile(FileLocator.java:46)
        at com.diffplug.spotless.maven.java.Eclipse.newFormatterStep(Eclipse.java:41)
        at com.diffplug.spotless.maven.FormatterFactory.lambda$newFormatter$1(FormatterFactory.java:74)
        at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
        at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
        at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
        at com.diffplug.spotless.maven.FormatterFactory.newFormatter(FormatterFactory.java:75)
        at com.diffplug.spotless.maven.AbstractSpotlessMojo.execute(AbstractSpotlessMojo.java:101)
        at com.diffplug.spotless.maven.AbstractSpotlessMojo.execute(AbstractSpotlessMojo.java:95)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:132)
        ... 20 more
Caused by: org.codehaus.plexus.resource.loader.FileResourceCreationException: Failed to create directory D:\git\project\api\target\https:\git.mycompany.de\projects\INFRA\repos\pm-codestyle\raw
        at org.codehaus.plexus.resource.DefaultResourceManager.createResourceAsFile(DefaultResourceManager.java:209)
        at org.codehaus.plexus.resource.DefaultResourceManager.getResourceAsFile(DefaultResourceManager.java:101)
        at com.diffplug.spotless.maven.FileLocator.locateFile(FileLocator.java:42)
        ... 34 more
```

So, i guess the problem is the colon ":" as well as the slashes "/" as they do not work on windows.

--- a/plugin-maven/CHANGES.md
+++ b/plugin-maven/CHANGES.md
@@ -2,6 +2,8 @@
 
 ### Version 1.10.0-SNAPSHOT - TBD ([javadoc](https://diffplug.github.io/spotless/javadoc/spotless-maven-plugin/snapshot/), [snapshot](https://oss.sonatype.org/content/repositories/snapshots/com/diffplug/spotless/spotless-maven-plugin/))
 
+* Fixed a bug in configuration file resolution on Windows when file is denoted by a URL. ([#254](https://github.com/diffplug/spotless/pull/254))
+
 ### Version 1.0.0.BETA5 - May 14th 2018 ([javadoc](https://diffplug.github.io/spotless/javadoc/spotless-maven-plugin/1.0.0.BETA5/), [jcenter](https://bintray.com/diffplug/opensource/spotless-maven-plugin/1.0.0.BETA5))
 
 * Fixed a bug in `LicenseHeaderStep` which caused an exception with some malformed date-aware licenses. ([#222](https://github.com/diffplug/spotless/pull/222))

--- a/plugin-maven/src/main/java/com/diffplug/spotless/maven/FileLocator.java
+++ b/plugin-maven/src/main/java/com/diffplug/spotless/maven/FileLocator.java
@@ -15,9 +15,10 @@
  */
 package com.diffplug.spotless.maven;
 
-import static com.diffplug.common.base.Strings.*;
+import static com.diffplug.common.base.Strings.isNullOrEmpty;
 
 import java.io.File;
+import java.util.UUID;
 
 import org.codehaus.plexus.resource.ResourceManager;
 import org.codehaus.plexus.resource.loader.FileResourceCreationException;
@@ -25,6 +26,8 @@ import org.codehaus.plexus.resource.loader.ResourceNotFoundException;
 import org.codehaus.plexus.util.FileUtils;
 
 public class FileLocator {
+
+	static final String TMP_RESOURCE_FILE_PREFIX = "spotless-resource-";
 
 	private final ResourceManager resourceManager;
 
@@ -48,9 +51,7 @@ public class FileLocator {
 	}
 
 	private static String tmpOutputFileName(String path) {
-		String nameWithExtension = FileUtils.filename(path);
 		String extension = FileUtils.extension(path);
-		String name = nameWithExtension.replace('.' + extension, "");
-		return name + '-' + System.currentTimeMillis() + '.' + extension;
+		return TMP_RESOURCE_FILE_PREFIX + UUID.randomUUID() + '.' + extension;
 	}
 }

--- a/plugin-maven/src/test/java/com/diffplug/spotless/maven/FileLocatorTest.java
+++ b/plugin-maven/src/test/java/com/diffplug/spotless/maven/FileLocatorTest.java
@@ -15,14 +15,13 @@
  */
 package com.diffplug.spotless.maven;
 
+import static com.diffplug.spotless.maven.FileLocator.TMP_RESOURCE_FILE_PREFIX;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNull;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.ArgumentMatchers.eq;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.*;
 
 import java.io.File;
 import java.nio.file.Paths;
@@ -47,17 +46,32 @@ public class FileLocatorTest {
 	}
 
 	@Test
-	public void locateValidFile() throws Exception {
-		String path = Paths.get("tmp", "configs", "my-config.xml").toString();
-		File tmpOutputFile = new File("tmp-my-config.xml");
+	public void locateXmlFile() throws Exception {
+		testFileLocator(Paths.get("tmp", "configs", "my-config.xml").toString(), "xml");
+	}
+
+	@Test
+	public void locatePropertiesFile() throws Exception {
+		testFileLocator(Paths.get("home", "ubuntu", "my-other-config.properties").toString(), "properties");
+	}
+
+	@Test
+	public void locateConfFileWithIncorrectSeparators() throws Exception {
+		String oppositeSeparator = "/".equals(File.separator) ? "\\" : "/";
+		String path = "tmp" + oppositeSeparator + "configs" + oppositeSeparator + "hello.conf";
+
+		testFileLocator(path, "conf");
+	}
+
+	private void testFileLocator(String path, String extension) throws Exception {
+		File tmpOutputFile = new File("tmp-file");
 		when(resourceManager.getResourceAsFile(any(), any())).thenReturn(tmpOutputFile);
 
 		File locatedFile = fileLocator.locateFile(path);
-
 		assertEquals(tmpOutputFile, locatedFile);
 
 		ArgumentCaptor<String> argCaptor = ArgumentCaptor.forClass(String.class);
 		verify(resourceManager).getResourceAsFile(eq(path), argCaptor.capture());
-		assertThat(argCaptor.getValue()).startsWith("my-config").endsWith(".xml");
+		assertThat(argCaptor.getValue()).startsWith(TMP_RESOURCE_FILE_PREFIX).endsWith('.' + extension);
 	}
 }

